core/src/main/java/org/apache/calcite/rel/rules/SortRemoveConstantKeysRule.java

```diff
@@ -44,7 +45,7 @@
 
   private SortRemoveConstantKeysRule() {
```

Comment:
Thanks a lot ~
Besides all rules in existing change (FilterProjectTransposeRule, ProjectMergeRule, SortRemoveConstantKeyRule), shall we include more rules, which only need to be applied for logical nodes ?

Reply:
I am confident that the rules I update here will not cause the change of search space, aka the transformation on physical nodes would result in an equivalent, so they are safe to remove. But we need to be careful updating other rules. And it'd better be done by the rule author. This change for now would solve most of the problems, if not all, and we can always come back to update other rules when feel necessary.

Reply:
Thanks for explain ~